koru-processor/src/jvmMain/kotlin/com/futuremind/koru/processor/WrapperClassBuilder.kt

```diff
@@ -145,26 +147,24 @@ class WrapperClassBuilder(
         fun TypeSpec.containsPropertySignature() =
             this.propertySpecs.any { it.hasSameSignature(this@overridesGeneratedInterface) }
 
-        return originalToGeneratedInterface?.generated?.typeSpec?.containsPropertySignature() == true
+        return generatedInterfaces.values.any { it.typeSpec.containsPropertySignature() }
     }
 
     /**
-     * 1. Add all original superinterfaces.
-     * 2. (optionally) Replace original superinterface if it's a standalone interface generated by @ToNativeInterface
-     * 3. (optionally) Add the superinterface autogenerated from current class with @ToNativeInterface
+     * 1. Replace original superinterfaces if they're standalone interfaces generated by @ToNativeInterface
+     * 2. Add the missing superinterface autogenerated from current class with @ToNativeInterface
      */
     private val superInterfaces: MutableList<TypeName> = originalTypeSpec.superinterfaces.keys
-        .map { interfaceName ->
-            when (originalToGeneratedInterface?.originalName == interfaceName) {
-                false -> interfaceName
-                true -> originalToGeneratedInterface!!.generated.name
+        .mapNotNull { interfaceName ->
+            when (val matchingSuper = generatedInterfaces.keys.find { it == interfaceName }) {
+                null -> null
+                else -> generatedInterfaces[matchingSuper]!!.name
             }
         }
         .toMutableList()
         .apply {
-            if (originalTypeName == originalToGeneratedInterface?.originalName) {
-                add(originalToGeneratedInterface.generated.name)
-            }
+            val missingSuperInterface = generatedInterfaces.values.map { it.name }.toSet().minus(this).elementAtOrNull(0)
```

Comment:
This is not correct - we are dealing with more than one set of generated interfaces - this will just randomly assign an interface from the general list of generated interfaces, not just the ones associated with this class.